mlflow/entities/span_event.py

```diff
@@ -16,14 +18,20 @@ class SpanEvent(_MlflowObject):
     Args:
         name: Name of the event.
         timestamp:  The exact time the event occurred, measured in microseconds.
+            If not provided, the current time will be used.
         attributes: A collection of key-value pairs representing detailed
             attributes of the event, such as the exception stack trace.
+            Attributes value must be one of ``[str, int, float, bool, bytes]``
+            or a sequence of these types.
     """
 
     name: str
-    # Use current time if not provided.
-    timestamp: Optional[int] = field(default=int(time.time() * 1e6))
-    attributes: Dict[str, Any] = field(default_factory=dict)
+    # Use current time if not provided. We need to use default factory otherwise
+    # the default value will be fixed to the build time of the class.
+    timestamp: Optional[int] = field(default_factory=lambda: int(time.time() * 1e6))
+    attributes: Dict[str, Union[_AttrValueType, Sequence[_AttrValueType]]] = field(
+        default_factory=dict
+    )
```

Comment:
Can `attributes` a nested dict like `{"a": {"b": "c"}}`?

Reply:
No, dictionary is not supported type for event attribute value.